File: frontend/src/models/user/TournamentParticipant.ts
export default class TournamentParticipant {
  userId!: number;
  name!: string;
  username!: string;
  answered!: boolean;
  score!: number;
  numberOfAnswered!: number;
  numberOfCorrect!: number;

  constructor(jsonObj?: TournamentParticipant) {
    if (jsonObj) {
      this.userId = jsonObj.userId;
      this.name = jsonObj.name;
      this.username = jsonObj.username;
      this.answered = jsonObj.answered;
      this.score = jsonObj.score;
      this.numberOfAnswered = jsonObj.numberOfAnswered;
      this.numberOfCorrect = jsonObj.numberOfCorrect;
    }
  }
}

File: backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/tournament/domain/TournamentParticipant.java
package pt.ulisboa.tecnico.socialsoftware.tutor.tournament.domain;

import javax.persistence.Column;
import javax.persistence.Embeddable;

@Embeddable
public class TournamentParticipant {

    @Column(name = "participant_id")
    private Integer id;

    @Column(name = "participant_username")
    private String username;

    @Column(name = "participant_name")
    private String name;

    @Column(name = "participant_answered")
    private boolean answered;

    public TournamentParticipant() {
    }

    public TournamentParticipant(Integer userId, String username, String name, boolean answered) {
        this.id = userId;
        this.username = username;
        this.name = name;
        this.answered = answered;
    }

    public Integer getId() {
        return id;
    }

    public String getUsername() {
        return username;
    }

    public String getName() {
        return name;
    }

    public boolean isAnswered() {
        return answered;
    }
}
